server/src/main/java/org/elasticsearch/common/lucene/search/MultiPhrasePrefixQuery.java

```diff
@@ -97,7 +97,7 @@ public void add(Term[] terms) {
      */
     public void add(Term[] terms, int position) {
         for (int i = 0; i < terms.length; i++) {
-            if (terms[i].field() != field) {
+            if (!Objects.equals(terms[i].field(), field)) {
```

Comment:
small styling issue: we prefer to make negative statements more explicit by using "v == false" instead of "!v". Even though it is more verbose it helps readability. So in this case "Objects.equals(terms[i].field(), field) == false" would be preferable.

Reply:
I've done it